tap_klaviyo/utils.py

```diff
@@ -3,13 +3,67 @@
 import singer
 from singer import metrics
 import requests
+import backoff
+import json
+import simplejson
 
 DATETIME_FMT = "%Y-%m-%dT%H:%M:%SZ"
 
-
 session = requests.Session()
 logger = singer.get_logger()
 
+class KlaviyoError(Exception):
+    pass
+
+class KlaviyoNotFoundError(KlaviyoError):
+    pass
+
+class KlaviyoBadRequestError(KlaviyoError):
+    pass
+
+class KlaviyoUnauthorizedError(KlaviyoError):
+    pass
+
+class KlaviyoForbiddenError(KlaviyoError):
+    pass
+
+class KlaviyoInternalServiceError(KlaviyoError):
+    pass
+
+ERROR_CODE_EXCEPTION_MAPPING = {
+    400: {
+        "raise_exception": KlaviyoBadRequestError,
+        "message": "Request is missing or has a bad parameter."
+    },
+    401: {
+        "raise_exception": KlaviyoUnauthorizedError,
+        "message": "Invalid authorization credentials."
+    },
+    403: {
+        "raise_exception": KlaviyoForbiddenError,
+        "message": "Request is missing or has an invalid API key."
+    },
+    404: {
+        "raise_exception": KlaviyoNotFoundError,
+        "message": "The requested resource doesn't exist."
+    },
+    500: {
+        "raise_exception": KlaviyoInternalServiceError,
+        "message": "Something is wrong on Klaviyo's end."
+    }
+}
+
+def raise_for_error(response):   
+    try:
+        json_resp = response.json()
```

Comment:
@harshpatel4crest Can you please get the response of response.raise_for_status() here and except the HTTTError in the exception.
Please refer to the tap-Xero code (https://github.com/singer-io/tap-xero/pull/85/files#diff-43d32c99c6992e01f4dbdb49df5e372db029dae0de306b9191283ef21b852438R216) for the same